common/update-probeconfig.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+
+# add/merge ProbeConfig.add into ProbeConfig's @PROBE_SPECIFIC_DATA@
+# updating any existing attributes
+
+import sys
+import os
+import re
+
+
+def read_if_exists(path):
+    try:
+        return open(path).read()
+    except IOError:
+        return ''
+
+
+def filter_nonlast_lines(lines):
+    haveattrs = set()
+    for line in lines:
+        m = re.match(r'^ *(\w+)=', line)
+        if m:
+            attr = m.group(1)
+            if attr in haveattrs:
+                continue
+            haveattrs.add(attr)
+        yield line
+
+
+def main(pc, pc_add, pc_out=None):
+    add_txt = read_if_exists(pc_add)
+    pc_txt = open(pc).read().replace('@PROBE_SPECIFIC_DATA@', add_txt)
+
+    lines = reversed(pc_txt.splitlines())
+    lines = list(filter_nonlast_lines(lines))
+    with open(pc_out or pc, "w") as outf:
+        print('\n'.join(reversed(lines)), file=outf)
```

Comment:
wat. Comment here would be helpful

Reply:
> wat. Comment here would be helpful

"Left as an exercise to the reader."

Wat, didn't you see the title of this book?

_Python Puzzles_
_Comments Confuse_
_Lambdas Lament_
_Unless You Defuse_

Reply:
I've added comments and rejiggered this a bit to attempt to improve readability.

Reply:
@brianhlin i know you've already pre-approved, but i'm throwing this back to you just to see if you are satisfied with the comments / (hopefully improved) readability updates for [update-probeconfig.py](https://github.com/edquist/gratia-probe/blob/SOFTWARE-4621.ProbeConfig-cleanup/common/update-probeconfig.py).

Reply:
Yeah, that helps. Thanks!